iam/__tests__/githubForkedRepoProvider.test.ts

```diff
@@ -0,0 +1,243 @@
+// ---- Test subject
+import { ForkedGithubRepoProvider } from "../src/providers/githubForkedRepoProvider";
+
+// ----- Types
+import { RequestPayload } from "@gitcoin/passport-types";
+
+// ----- Libs
+import axios from "axios";
+
+jest.mock("axios");
+
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+const validGithubUserResponse = {
+  data: {
+    id: 18723656,
+    login: "a-cool-user",
+    type: "User",
+  },
+  status: 200,
+};
+
+const validGithubUserRepoResponse = {
+  data: [
+    {
+      owner: {
+        id: 18723656,
+        type: "User",
+      },
+      fork: false,
+      forks_count: 5,
+    },
+    {
+      owner: {
+        id: 18723656,
+        type: "User",
+      },
+      fork: true,
+      forks_count: 2,
+    }
+  ],
+  status: 200,
+};
+
+const invalidGithubUserRepoResponse = {
+  data: [
+    {
+      owner: {
+        id: 18723656,
+        type: "User",
+      },
+      fork: false,
+      forks_count: 0,
+    },
+  ],
+  status: 200,
+};
+
+const validCodeResponse = {
+  data: {
+    access_token: "762165719dhiqudgasyuqwt6235",
+  },
+  status: 200,
+};
+
+const code = "ABC123_ACCESSCODE";
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  mockedAxios.post.mockImplementation(async (url, data, config) => {
+    return validCodeResponse;
+  });
+  
+  mockedAxios.get.mockImplementation(async (url, config) => {
+    if (url.endsWith("/user")) {
+      return validGithubUserResponse;
+    }
+    if (url.endsWith("/repos")) {
+      return validGithubUserRepoResponse;
+    }
+  });
+});
+
+describe("Attempt verification", function () {
+  it("handles valid verification attempt", async () => {
```

Comment:
Also please create tests for the edge cases:
  - a user has only a repos with fork_count=1 and he is the owner of the fork
  - a user has only a repos with fork_count=1 and he is not owner of the fork

Reply:
For the forks, a user can't fork their own repo 🙌